CRM/Resource/DAO/ResourceDemand.php

```diff
@@ -113,6 +113,17 @@ public static function getReferenceColumns() {
     return Civi::$statics[__CLASS__]['links'];
   }
 
+  /**
+   * CAREFUL: DON'T DELETE THIS WHEN REBUILDING THE DAOs! */
```

Comment:
Great work, @CsarRamos, but shouldn't there be a way to define this sort of thing in [the schema](https://github.com/systopia/de.systopia.resource/blob/master/xml/schema/CRM/Resource/ResourceDemand.xml) so we wouldn't have to worry about the DAO rebuild? @colemanw, would you happen to know this off the top of your head?

Reply:
@bjendres yes - `buildOptions` funtions belong in the BAO file, not the DAO. Just move it there and all should be good.

Reply:
Thanks! That's exactly what I wanted to hear :-)

Reply:
Thanks for the help, I tried it but it didn't work, the **$fieldName** in `buildOptions` return `resource_type_id` instead of `entity_table` when compose join in searchkit. I tried to replicate the same logic on Resource class: https://github.com/systopia/de.systopia.resource/blob/master/CRM/Resource/DAO/Resource.php#L99
maybe I need to rethink the approach and modify schema. Any suggestions are welcome, thanks!